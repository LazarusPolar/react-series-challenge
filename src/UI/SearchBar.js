import React from 'react';
import { StyledSearchbar, StyledInput, StyledButton } from './Searchbar.styled';

const SearchBar = (props) => {
    return (
        <StyledSearchbar>
            <StyledInput 
            type="text" 
            placeholder = "Search for awesome gifs" 
            onChange = {(event) => {props.updateState(event.target.value)}} />

            <StyledButton onClick = { props.handleSearch } />  
        </StyledSearchbar>
    );
}

export default SearchBar;