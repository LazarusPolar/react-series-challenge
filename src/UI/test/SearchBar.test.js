import React, {Fragment} from 'react';
import SearchBar from '../SearchBar'
import { StyledSearchbar, StyledInput, StyledButton } from '../Searchbar.styled';
import { shallow, mount, render } from 'enzyme';

describe('SearchBar', () => {
    const func = jest.fn();
    const func1 = jest.fn();
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <SearchBar 
            updateState = { func } 
            handleSearch = { func1 } />
        );
    });

    test('should set SearchBar ', () => {
        const comp = render(
            <StyledSearchbar>
                <StyledInput 
                type="text" 
                placeholder = "Search for awesome gifs"
                onChange = { func } />

                <StyledButton onClick = { func1 } />  
            </StyledSearchbar>)

        expect(comp).toHaveLength(1);
    });

    test('should not set SearchBar ', () => {
        expect(wrapper.contains(
            <StyledSearchbar>
                <StyledInput 
                type="text" 
                placeholder = "Search for awesome gifs"
                onChange = { func } />

                <StyledButton onClick = { func1 } />  
            </StyledSearchbar>
        )).toBe(false);
    });

})
