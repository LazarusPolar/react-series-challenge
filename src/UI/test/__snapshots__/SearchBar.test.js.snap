// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    handleSearch={[Function]}
    updateState={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.input
          onChange={[Function]}
          placeholder="Search for awesome gifs"
          type="text"
/>,
        <styled.button
          onClick={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onChange": [Function],
          "placeholder": "Search for awesome gifs",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.input
            onChange={[Function]}
            placeholder="Search for awesome gifs"
            type="text"
/>,
          <styled.button
            onClick={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Search for awesome gifs",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
